src/main/java/org/spongepowered/api/item/recipe/Recipe.java

```diff
@@ -47,21 +46,21 @@
 public interface Recipe {
 
     /**
-     * Returns the list of item types that result when successful crafting of
+     * Returns the list of item stacks that result when successful crafting of
      * this Recipe is completed.
      *
      * @return The resultant list of item types
      */
-    List<ItemType> getResultTypes();
+    List<ItemStack> getResults();
 
     /**
      * Checks if the given {@link GridInventory} fits the required constraints
      * to craft this Recipe.
      *
-     * @param grid The ItemGrid to check for validity
+     * @param grid The grid inventory to match against
      * @return True if the given input matches this recipe's requirements
      */
-    boolean isValid(GridInventory grid);
+    boolean matches(GridInventory grid);
```

Comment:
Maybe this could extend `Predicate<GridInventory>` (and then `@Override` the test method and add the javadocs using that override)?